packages/core/src/account/base.ts

```diff
@@ -328,6 +334,21 @@ export abstract class BaseSmartContractAccount<
     return [factoryAddress, factoryCalldata];
   }
 
+  protected async getImplementationAddress(): Promise<"0x0" | Address> {
+    const accountAddress = await this.getAddress();
+
+    const storage = await this.rpcProvider.getStorageAt({
+      address: accountAddress,
+      slot: "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc",
```

Comment:
maybe worth humanizing this with a comment?

Reply:
yup can do